src/newshop.cpp

```diff
@@ -1359,11 +1361,19 @@ void shop_list(char *arg, struct char_data *ch, struct char_data *keeper, vnum_t
       else strcpy(buf2, "-");
 
       if (IS_OBJ_STAT(obj, ITEM_NERPS)) {
-        snprintf(ENDOF(buf), sizeof(buf) - strlen(buf), "^Y(N)^n %-29s^n %-6s%2s   %0.2f%c  %9d\r\n", GET_OBJ_NAME(obj),
+        //Format string: "^Y(N)^n %-58s^n %-6s%2s   %0.2f%c  %9d\r\n"
+        //We apply padding for color codes here.
+        snprintf(paddingnumberstr, sizeof(paddingnumberstr), "%d", 58 + count_color_codes_in_string(GET_OBJ_NAME(obj)));
+        snprintf(formatstr, sizeof(formatstr), "%s%s%s", "^Y(N)^n %-", paddingnumberstr, "s^n %-6s%2s   %0.2f%c  %9d\r\n");
+        snprintf(ENDOF(buf), sizeof(buf) - strlen(buf), formatstr, GET_OBJ_NAME(obj),
                 GET_OBJ_TYPE(obj) == ITEM_CYBERWARE ? "Cyber" : "Bio", buf2, ((float)GET_OBJ_VAL(obj, 4) / 100),
                 GET_OBJ_TYPE(obj) == ITEM_CYBERWARE ? 'E' : 'I', buy_price(obj, shop_nr));
       } else {
-        snprintf(ENDOF(buf), sizeof(buf) - strlen(buf), "%-33s^n %-6s%2s   %0.2f%c  %9d\r\n", GET_OBJ_NAME(obj),
+        //Format string: "%-62s^n %-6s%2s   %0.2f%c  %9d\r\n"
+        //We apply padding for color codes here.
+        snprintf(paddingnumberstr, sizeof(paddingnumberstr), "%d", 62 + count_color_codes_in_string(GET_OBJ_NAME(obj)));
```

Comment:
62 is also magical! As are 46, 50, 30, 50, and 50 below.

Reply:
Reviewing more, I see that you're pulling the numbers from the old format strings... it looks like you get the unenviable task of figuring out what the formula was for generating those numbers in the first place and then figuring out how to make that readable here. I suspect it was derived from the length of the dash padding in the tables, but I'm not certain.

Reply:
As I said above, indeed I am deriving from the existing numbers. I am pretty sure the original numbers were not calculated algorithmically. They look like they were arbitrary chosen to fit current-at-the-time strings/dashes/formatting and were possibly extended over time for longer strings. Evidence on that is that tables between different types of stores do not match. Width per store type is as required for strings to fit. There's really nothing that can be done about this except, as I said above to start saving letters with abbreviations etc and that will only save us so much. But the question remains, is it worth conforming to a limitation that doesn't really apply nowadays, specially on strings that are nowhere else limited in the code (eg not passed to arrays with fixed short lengths etc)?

As a slightly related sidenote, I want to add more variables to prompt (a lot of them requested, eg progress) and largely enlarge the limit as, at least in the prompt's case, it's a string that only you can see and you should be able to set it as long as you want it to, even making it multiline with a new line variable. Could make nice color coded prompts with lots of information on them that currently don't fit and you're not forcing it on anyone else so I don't see why not.

Reply:
The magic numbers bit isn't about shortening lines, it's just about making the code maintainable going forward. The solution I'm advocating for is a two-parter: first, pick the longest line in the output (usually the padding dashes at the top) and split it out into its own static string that you concatenate into the display, then replace the magic numbers with `strlen(the_new_string)`. That makes it clear what the padding is based off of, and also automatically updates the padding if we extend the string later.

Now, if that number was wholly arbitrary to begin with, or was based on something dumb like subtracting the total possible length of other fields from the length of the dash-line or something... eh. It should still be done, but the effort to do it becomes more than I'm willing to put in right now.

Prompts can be extended, it's just currently limited by the SQL field's length, so you'd need a migration for that and some code in db.cpp to check the migration was applied. Also, there's already a newline variable in prompts, so you wouldn't need to add that.

Reply:
Ahhh gotcha. Yeah we can do that. It would probably mean splitting the top string into per column parts if we want to strlen() it to calculate the paddings for each section. I wonder if I should just replace those calls entirely with std::stringstream instead of using C-stryle formatting strings there. If I'm gonna rewrite it, might as well bring it to C++ and this century in the process.

I had not checked existing variables yet but yes but I've seen the SQL type for prompt is set to varchar(79) while working on the obj order saving stuff.

Reply:
I will rewrite this part in a later PR. As currently working on the one changing object ordering is making this branch a hassle to test as I have to dump and restore dbs continuously to change branches.